test/data/biosample_test.json

```diff
@@ -105,7 +105,7 @@
       "sample_collection_site": "Lithgow State Coal Mine"
     },
     {
-      "id": "nmdc:e924072f-98b5-4f88-a796-a7ba1d8ddd92",
+      "id": "nmdc:bsm-bsm-e924072f-98b5-4f88-a796-a7ba1d8ddd92",
```

Comment:
Why does this have two typecodes?

Reply:
Good catch. The only direct validation I've put in place yet is `^nmdc:{typecode}`.  I have added the entire `id` pattern for each class in a `strucutred_pattern`, but I don't believe the support for expanding them is complete in LinkML yet. I will check tomorrow.

Bottom line, the test `id` of "nmdc:bsm-bsm-e924072f-98b5-4f88-a796-a7ba1d8ddd92" passes because the plain `pattern` validation isn't anchored on the right yet.

I'll fix the example.